packages/desktop/src/renderer/sagas/socket/socket.saga.ts

```diff
@@ -29,24 +34,35 @@ function* handleSocketLifecycleActions(socket: Socket): Generator {
   })
 }
 
-function subscribeSocketLifecycle(socket: Socket) {
+function subscribeSocketLifecycle(socket?: Socket) {
   return eventChannel<
   ReturnType<typeof socketActions.setConnected> |
   ReturnType<typeof socketActions.suspendConnection>
   >(emit => {
-    socket.on('connect', async () => {
+    socket?.on('connect', async () => {
       console.log('websocket connected')
       emit(socketActions.setConnected())
     })
-    socket.on('disconnect', () => {
+    socket?.on('disconnect', () => {
       console.log('closing socket connection')
       emit(socketActions.suspendConnection())
     })
     return () => {}
   })
 }
 
-function* cancelRootTaskSaga(task: FixedTask<Generator>): Generator {
+function* cancelRootSaga(task: FixedTask<Generator>): Generator {
   console.log('canceling root task')
   yield* cancel(task)
 }
+
+function* cancelObservers(task: FixedTask<Generator>): Generator {
+  console.log('canceling observers')
```

Comment:
leftover?

Reply:
No, it's there in purpose. Similar to the one above it informs about action taken during detaching socket